cli/cmd/root.go

```diff
@@ -0,0 +1,138 @@
+// Package cmd has code for all the commands of fanotify-mon.
+package cmd
+
+import (
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/kinvolk/fanotify-poc/internal"
+	"github.com/kinvolk/fanotify-poc/pkg/containerd"
+	"github.com/kinvolk/fanotify-poc/pkg/docker"
+	"github.com/kinvolk/fanotify-poc/pkg/k8s"
+	containercollection "github.com/kinvolk/inspektor-gadget/pkg/container-collection"
+	"github.com/kinvolk/inspektor-gadget/pkg/gadgettracermanager/pubsub"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"golang.org/x/sys/unix"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+var (
+	hostname    string
+	hostRuntime string
+	kubeconfig  string
+)
+
+var RootCmd = &cobra.Command{
+	Use:   "fanotify-mon",
+	Short: "Monitor for fanotify",
+	Run: func(cmd *cobra.Command, args []string) {
+		fanotify(hostname, hostRuntime, kubeconfig)
+	},
+}
+
+func Execute() {
+	if err := RootCmd.Execute(); err != nil {
+		os.Exit(1)
+	}
+}
+
+func init() {
+	log.SetLevel(log.DebugLevel)
+	RootCmd.DisableAutoGenTag = true
+
+	pf := RootCmd.PersistentFlags()
+	pf.StringVarP(&hostname, "hostname", "", "", "Name of node in k8s control plane")
```

Comment:
It does not have to be control plane, it is just the name of the node on which this app is running.